src/main/java/com/aws/greengrass/deployment/DeviceConfiguration.java

```diff
@@ -218,6 +219,20 @@ public DeviceConfiguration(Kernel kernel, String thingName, String iotDataEndpoi
         validate();
     }
 
+    /**
+     * Constructor used for connectivity validation.
+     *
+     * @param currentConfig         current device configs
+     * @param deploymentConfig      Map of the configs being merged in from deployment
+     * @param deploymentTimestamp   timestamp of the deployment being merged in
+     */
+    public DeviceConfiguration(DeviceConfiguration currentConfig, Map<String, Object> deploymentConfig,
+                               long deploymentTimestamp) {
+        this.kernel = new Kernel(currentConfig, deploymentConfig, deploymentTimestamp);
```

Comment:
do not make a new kernel.

Reply:
the existing Kernel already has the current device configuration, what should I do if I don't make a new Kernel?

Reply:
You aren't creating a kernel just for fun, you're creating it presumably because you think you need it. So go back to that, why is it needed at all? How can you avoid needing it?